File: easy/Climbing_Stairs.cpp
//from : https://leetcode.com/problems/climbing-stairs/

class Solution {
public:
    int climbStairs(int n) {
        vector<int> v = {0, 1, 2};
        for(int i = 3; i<=n; i++)
            v.push_back(v[i-2] + v[i-1]);
        return v[n];
    }
};

File: PHP/Easy/Climbing_Stairs.php
//from : https://leetcode.com/problems/climbing-stairs/


<?php

class Solution {
    /**
     * @param Integer $n
     * @return Integer
     */
    function climbStairs($n) {
        $dp = array_fill(0, $n + 1, 0);
        $dp[0] = 1; //first step
        for($i = 0; $i < $n; $i++) {
            $dp[$i + 1] += $dp[$i];
            $dp[$i + 2] += $dp[$i];
        }
        return $dp[$n];
    }
}
